src/lib/fetch.ts

```diff
@@ -452,6 +466,108 @@ function sendZkappQuery(json: string) {
 `;
 }
 
+const getEventsQuery = (
+  publicKey: string,
+  tokenId: string,
+  filterOptions?: EventActionFilterOptions
+) => {
+  const { to, from } = filterOptions ?? {};
+  let input = `address: "${publicKey}", tokenId: "${tokenId}"`;
+  if (to !== undefined) {
+    input += `, to: ${to}`;
+  }
+  if (from !== undefined) {
+    input += `, from: ${from}`;
+  }
+  return `{
+  events(input: { ${input} }) {
+    blockInfo {
+      distanceFromMaxBlockHeight
+      height
+    }
+    eventData {
+      index
+      data
+    }
+  }
+}
+`;
+};
+
+type EventActionFilterOptions = {
+  to?: UInt32;
+  from?: UInt32;
+};
+
+/**
+Asynchronously fetches event data for an account from the Mina Archive Node GraphQL API.
+@async
+@param {object} accountInfo - The account information object.
+@param {string} accountInfo.publicKey - The account public key.
+@param {string} [accountInfo.tokenId] - The optional token ID for the account.
+@param {string} [graphqlEndpoint=archiveGraphqlEndpoint] - The GraphQL endpoint to query. Defaults to the Archive Node GraphQL API.
+@param {object} [filterOptions={}] - The optional filter options object.
+@returns {Promise<Array>} A promise that resolves to an array of objects containing event data and block height for the account.
+@throws {Error} If the GraphQL request fails or the response is invalid.
+@example
+const accountInfo = { publicKey: 'B62qiwmXrWn7Cok5VhhB3KvCwyZ7NHHstFGbiU5n7m8s2RqqNW1p1wF' };
+const events = await fetchEvents(accountInfo);
+console.log(events);
+*/
+async function fetchEvents(
+  accountInfo: { publicKey: string; tokenId?: string },
+  graphqlEndpoint = archiveGraphqlEndpoint,
```

Comment:
The archive endpoint could be undefined, I think we should likely catch that as early as possible when someone calles `myContract.fetchEvents`

Reply:
Good call! Added some code to throw an error if not specified